enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/SdkSaveQuestionForm.tsx

```diff
@@ -6,14 +6,28 @@ import { SaveQuestionProvider } from "metabase/components/SaveQuestionForm/conte
 import { useCreateQuestion } from "metabase/query_builder/containers/use-create-question";
 import { useSaveQuestion } from "metabase/query_builder/containers/use-save-question";
 import { Stack, Title } from "metabase/ui";
+import type Question from "metabase-lib/v1/Question";
 
 import { useInteractiveQuestionContext } from "../context";
 
-export const SaveQuestion = ({ onClose }: { onClose: () => void }) => {
+export type SdkSaveQuestionFormProps = {
+  onSave?: (question?: Question) => Promise<void>;
+  onClose?: () => void;
+};
+
+export const SdkSaveQuestionForm = ({
+  onClose,
+  onSave,
+}: SdkSaveQuestionFormProps) => {
   const { question, originalQuestion } = useInteractiveQuestionContext();
 
   const handleCreate = useCreateQuestion();
-  const handleSave = useSaveQuestion();
+  const handleSaveQuestion = useSaveQuestion();
+
+  const handleSave = async (question: Question) => {
+    await onSave?.(question);
```

Comment:
Can you explain more why this is designed to be async? I typically see callback being just a typical function hence the name callback. We call the callback when something happens and we don't care about the result, so seeing `await` here makes me curious.

Reply:
Good eye - I think it can be synchronous. Another note is that the external `onSave` prop gets called before the `handleSaveQuestion` which would run before the question is saved.

Reply:
If we name this callback `onBeforeSave` that might make sense to make it async. But if it's `onSave` I'd just expect it to be called as is without using the result from the callback.

What do you think?

Reply:
the idea was that, since this is a user-facing function, that they might want to do something asynchronous with the question, whether that's a UI change or something with the data. I don't mind changing it to synchronous but I imagine an asynchronous function would provide a little flexibility to the end user

Reply:
so maybe onBeforeSave makes sense there, then `onSave` would be the final thing. But I think we should still pass the question. I'm imagining the scenario where a user might want to have a toast that says something like `Question 1 has been saved!` once `onSave` is triggered. They could still have the same access to the data

Reply:
I'm onboard with passing `question`. The new API looks good btw.